paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/PaymentMethodVerticalLayoutInteractor.kt

```diff
@@ -14,48 +20,105 @@ internal interface PaymentMethodVerticalLayoutInteractor {
     data class State(
         val supportedPaymentMethods: List<SupportedPaymentMethod>,
         val isProcessing: Boolean,
+        val selectedPaymentMethodIndex: Int,
     )
 
     sealed interface ViewAction {
         data object TransitionToManageSavedPaymentMethods : ViewAction
-        data class TransitionToForm(val selectedPaymentMethodCode: String) : ViewAction
+        data class PaymentMethodSelected(val selectedPaymentMethodCode: String) : ViewAction
     }
 }
 
 internal class DefaultPaymentMethodVerticalLayoutInteractor(
-    private val viewModel: BaseSheetViewModel,
+    paymentMethodMetadata: PaymentMethodMetadata,
```

Comment:
I wanted to be able to unit test this, so I separated it from the viewModel, but left a view model constructor to make things easy.